app/actions/remote/thread.ts

```diff
@@ -216,3 +222,152 @@ export const updateThreadFollow = async (serverUrl: string, teamId: string, thre
+    if (!currentUser) {
+        return {error: true};
+    }
+
+    const {config} = await getCommonSystemValues(operator.database);
+    const data: Thread[] = [];
+
+    const fetchThreads = async (opts: FetchThreadOptions) => {
+        let page = 0;
+        const {before, after, perPage = General.CRT_CHUNK_SIZE, deleted, unread, since} = opts;
+
+        page += 1;
+        const {threads} = await client.getThreads(config.Version, currentUser.id, teamId, before, after, perPage, deleted, unread, since);
+        if (threads.length) {
+            // Mark all fetched threads as following
+            threads.forEach((thread: Thread) => {
+                thread.is_following = true;
+                if (!unread) {
+                    thread.loaded_in_global_threads = true;
+                }
+            });
+
+            data.push(...threads);
+
+            if (threads.length === perPage) {
+                const newOptions: FetchThreadOptions = {perPage, deleted, unread};
+                if (direction === Direction.Down) {
+                    const last = threads[threads.length - 1];
+                    newOptions.before = last.id;
+                } else {
+                    const first = threads[0];
+                    newOptions.after = first.id;
+                }
+                if (pages != null && page < pages) {
+                    fetchThreads(newOptions);
+                }
+            }
+        }
+    };
+
+    try {
+        await fetchThreads(options);
+    } catch (error) {
+        if (__DEV__) {
+            throw error;
+        }
+
+        return {error, data};
+    }
+
+    return {error: false, data};
+}
+
+export async function fetchNewThreads(
+    serverUrl: string,
+    teamId: string,
+    prepareRecordsOnly = false,
+): Promise<{error: unknown; models?: Model[]}> {
+    const options: FetchThreadOptions = {
+        unread: false,
+        deleted: true,
+        perPage: 60,
+    };
+
+    const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
+
+    if (!operator) {
+        return {error: `${serverUrl} database not found`};
+    }
+
+    const newestThread = await getNewestThreadInTeam(operator.database, teamId, false);
+    options.since = newestThread ? newestThread.lastReplyAt : 0;
+
+    let response = {
+        error: undefined,
+        data: [],
+    } as {
+        error: unknown;
+        data?: Thread[];
+    };
+
+    // if we have no threads in the DB fetch all unread ones
+    if (options.since === 0) {
+        // options to fetch all unread threads
+        options.deleted = false;
+        options.unread = true;
+    }
+
+    response = await fetchBatchThreads(serverUrl, teamId, options);
+
+    const {error: nErr, data} = response;
+
+    if (nErr) {
+        return {error: nErr};
+    }
+
+    if (!data?.length) {
+        return {error: false, models: []};
+    }
+
+    const {error, models} = await processReceivedThreads(serverUrl, data, teamId, true);
```

Comment:
there are changes requested in #6075 that will affect this PR or if this PR is merged, then #6075 will need to consider this additions.

Reply:
We can treat PRs independently and review the changes again to remove the dependency as these two PRs are getting merged into different branches?

Reply:
@anurag6713 I understand your point but I do see that approach causing the following:
1. Potential conflict resolution
2. More reviews than needed
3. Confusion into what made it to one branch and what made it to another.

I know you guys are working in a feature branch that I'm even afraid to look at, that is yet another reason why I would continue encouraging you guys to make smaller PR's to the actual `gekidou` branch to avoid having to work twice on the same thing, resolve conflicts multiple times and review the same thing over and over again.